utils/redis.ts

```diff
@@ -57,24 +57,25 @@ export const createClusterRedisClient = async (
   return redisClient;
 };
 
-export const REDIS_CLIENT = pipe(
-  config.isProduction,
-  O.fromPredicate<boolean>(identity),
-  O.chainNullableK(_ => config.REDIS_CLUSTER_ENABLED),
-  O.chain(O.fromPredicate(identity)),
-  O.map(() =>
-    createClusterRedisClient(
-      config.REDIS_URL,
-      config.REDIS_PASSWORD,
-      config.REDIS_PORT
+export const redisClientFactory = (): Promise<RedisClient> =>
```

Comment:
Possible improvement: we can turn it into a singleton so that every request share the same connection pool instance:

```ts
let redisClient: RedisClient;
export const redisClientFactory = (): Promise<RedisClient> => {
   if(!redisClient){
     redisClient = // here we build the client
   }
   return redisClient
}
  
```

Reply:
I exposed a singleton by using a class in https://github.com/pagopa/io-functions-cgn/pull/102/commits/1e5923a1d4661210ca8ea4767ed4016fd7ed2481